dev-docs/RFCs/v5.0/viewport-animation-rfc.md

```diff
@@ -27,13 +27,13 @@ When rendering deck.gl layers on top of Mapbox map.
 ```js
       <AnimationMapController
```

Comment:
This render structure should not be required. `AnimationMapController` should replace `MapController` in both deck.gl and react-map-gl, and the old structure:
```
<MapGL {...viewport}>
    <DeckGL {...viewport} />
</MapGL>
```
will suffice unless deck.gl is used without map.

Reply:
While I am supportive of the idea to merge `Controller` into `DeckGL`, for the animation to be able to affect the map, the order will need to be reversed:
```
<DeckGL ...>
   <MapGL ...>
</DeckGL>
```
Unless we duplicate the entire animation functionality into react-map-gl of course, which is still being discussed.

This is due to animation being done by modifying child props, which is required to avoid initial "flashes" in animation.

Reply:
It is one of the top priorities for react-map-gl to support viewport animation, so that is definitely happening.